File: lib/src/main/java/com/yalantis/contextmenu/lib/HesitateInterpolator.java
package com.yalantis.contextmenu.lib;

import android.view.animation.Interpolator;

public class HesitateInterpolator implements Interpolator {

    public HesitateInterpolator() {
    }

    public float getInterpolation(float t) {
        float x = 2.0f * t - 1.0f;
        return 0.5f * (x * x * x + 1.0f);
    }
}

File: lib/src/main/java/com/yalantis/contextmenu/lib/HesitateInterpolator.kt
package com.yalantis.contextmenu.lib

import android.view.animation.Interpolator

class HesitateInterpolator : Interpolator {

    override fun getInterpolation(input: Float): Float {
        val x = 2.0f * input - 1.0f
        return 0.5f * (x * x * x + 1.0f)
    }
}